cmd/nodecmd/create.go

```diff
@@ -0,0 +1,678 @@
+	ux.Logger.PrintToUser("===========END OF FILE===========")
+	ux.Logger.PrintToUser("More info can be found at https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html#file-format-creds")
+}
+
+func createNode(_ *cobra.Command, args []string) error {
+	clusterName := args[0]
+
+	err := removeExistingTerraformFiles()
+	if err != nil {
+		return err
+	}
+	usr, _ := user.Current()
+	keyPairName := usr.Username + "-avalanche-cli"
+	certName := keyPairName + "-kp.pem"
+	securityGroupName := keyPairName + "-sg"
+	region := "us-east-2"
+	ami := "ami-0430580de6244e02e"
+
+	hclFile := hclwrite.NewEmptyFile()
+	tfFile, err := os.Create("node_config.tf")
+	if err != nil {
+		return err
+	}
+	rootBody := hclFile.Body()
+	creds := credentials.NewSharedCredentials("", "default")
+	credValue, err := creds.Get()
+	if err != nil {
+		printNoCredentialsOutput()
+		return err
+	}
+	err = setCloudCredentials(rootBody, credValue.AccessKeyID, credValue.SecretAccessKey, region)
+	if err != nil {
+		return err
+	}
+	// Load session from shared config
+	sess, err := session.NewSession(&aws.Config{
+		Region:      aws.String(region),
+		Credentials: creds,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Create new EC2 client
+	ec2Svc := ec2.New(sess)
+	var useExistingKeyPair bool
+	keyPairExists, err := checkKeyPairExists(ec2Svc, keyPairName)
+	if err != nil {
+		return err
+	}
+	if !keyPairExists {
+		setKeyPair(rootBody, keyPairName, certName)
+	} else {
+		certFilePath, err := getCertFilePath(certName)
+		if err != nil {
+			return err
+		}
+		if checkCertInSSHDir(certFilePath) {
+			useExistingKeyPair = true
+		} else {
+			ux.Logger.PrintToUser(fmt.Sprintf("Default Key Pair named %s already exists", keyPairName))
+			keyPairName, err = getNewKeyPairName(ec2Svc)
+			if err != nil {
+				return err
+			}
+			certName = keyPairName + "-kp.pem"
+			setKeyPair(rootBody, keyPairName, certName)
+		}
+	}
+	securityGroupExists, sg, err := checkSecurityGroupExists(ec2Svc, securityGroupName)
+	if err != nil {
+		return err
+	}
+	userIPAddress, err := getIPAddress()
+	if err != nil {
+		return err
+	}
+	if !securityGroupExists {
+		setSecurityGroup(rootBody, userIPAddress, securityGroupName)
+	} else {
+		ipInTCP, ipInHTTP := checkCurrentIPInSg(sg, userIPAddress)
+		setSecurityGroupRule(rootBody, userIPAddress, *sg.GroupId, ipInTCP, ipInHTTP)
+	}
+	setElasticIP(rootBody)
+	setUpInstance(rootBody, securityGroupName, useExistingKeyPair, keyPairName, ami)
+	setOutput(rootBody)
+	_, err = tfFile.Write(hclFile.Bytes())
+	if err != nil {
+		return err
+	}
+	instanceID, elasticIP, err := runTerraform()
+	if err != nil {
+		return err
+	}
+	certFilePath, err := getCertFilePath(certName)
+	if err != nil {
+		return err
+	}
+
+	if !useExistingKeyPair {
+		err = handleCerts(certName)
+		if err != nil {
+			return err
+		}
+	}
+
+	inventoryPath := "inventories/" + clusterName
```

Comment:
constant, app package

Reply:
inside .avalanche-cli

Reply:
could you move it to temporary and set cmd.Dir